src/main/java/duke/dukehelper/CommandHandler.java

```diff
@@ -0,0 +1,364 @@
+/**
+ * Class to handle command inputs by user
+ */
+package duke.dukehelper;
+import duke.taskhelper.ListTasks;
+import duke.taskhelper.Packet;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import static duke.dukehelper.Constants.Error.NO_ERROR;
+import static duke.dukehelper.Routine.replyRoutine;
+
+public class CommandHandler {
+    public static ListTasks list = new ListTasks();
+    private static SaveManager saveManager = null;
+    private static SaveManager loadManager = null;
+    private static int conversationCounter = 0;
+    /**
+     * Validates if the input supplies a description as needed by the command
+     * @param packet
+     * @throws DukeException.InvalidDescription
+     */
+    private static void validatePayload(Packet packet) throws DukeException.InvalidDescription {
+        if (packet.getPacketPayload() == null){
+            throw new DukeException.InvalidDescription();
+        }
+    }
+
+    /**
+     * Validates if list is empty
+     * @throws DukeException.NoList
+     */
+    private static void validateList() throws DukeException.NoList {
```

Comment:
Such a method to see whether the list is empty might be better placed in the ListTasks class definition..

Reply:
There is already such a function in list task, but I wanted the function to be in CommandHandler for smoother error handling. Any thoughts?